test/httpoison_base_test.exs

```diff
@@ -50,20 +39,6 @@ defmodule HTTPoisonBaseTest do
     assert validate :hackney
   end
 
-  test "request body using ExampleDefp" do
```

Comment:
It looks like it does not work with behaviours to mix def and defp
and results in warnings like:

```
warning: function process_request_headers/1 required by behaviour HTTPoison.Base was implemented as "defp" but should have been "def" (in module HTTPoisonBaseTest.ExampleDefp)
    test/httpoison_base_test.exs:16
```